linode/helper/framework_data.go

```diff
@@ -17,6 +18,19 @@ func KeepOrUpdateBool(original types.Bool, updated bool, preserveKnown bool) typ
 	return KeepOrUpdateValue(original, types.BoolValue(updated), preserveKnown)
 }
 
+func KeepOrUpdateSet(
```

Comment:
Can we take in an `elementType` param to make this func more generic? Currently it only resolves a set of string and it hopefully can be used for sets with other types of elements.